src/dbnode/storage/series/reader.go

```diff
@@ -203,6 +203,52 @@ func (r Reader) readersWithBlocksMapAndBuffer(
 	return results, nil
 }
 
+// FetchIndexChecksum reads index checksum blocks using just a block retriever.
+func (r Reader) FetchIndexChecksum(
+	ctx context.Context,
+	blockStart time.Time,
+	nsCtx namespace.Context,
+) (ident.IndexChecksum, error) {
+	return r.fetchIndexChecksum(ctx, blockStart, nsCtx)
+}
+
+func (r Reader) fetchIndexChecksum(
+	ctx context.Context,
+	blockStart time.Time,
+	nsCtx namespace.Context,
+) (ident.IndexChecksum, error) {
+	var (
+		nowFn        = r.opts.ClockOptions().NowFn()
+		now          = nowFn()
+		ropts        = r.opts.RetentionOptions()
+		size         = ropts.BlockSize()
+		alignedStart = blockStart.Truncate(size)
+	)
+	// Squeeze the lookup window by what's available to make range queries like [0, infinity) possible
+	earliest := retention.FlushTimeStart(ropts, now)
+	if alignedStart.Before(earliest) {
+		alignedStart = earliest
+	}
+
+	if r.retriever == nil {
+		return ident.IndexChecksum{}, nil
+	}
+	// Try to stream from disk
+	isRetrievable, err := r.retriever.IsBlockRetrievable(alignedStart)
+	if err != nil {
+		return ident.IndexChecksum{}, err
+	} else if !isRetrievable {
+		return ident.IndexChecksum{}, nil
+	}
+	streamedBlock, err := r.retriever.StreamIndexChecksum(ctx,
+		r.id, alignedStart, nsCtx)
+	if err != nil {
+		return ident.IndexChecksum{}, err
+	}
+
+	return streamedBlock.RetrieveIndexChecksum()
```

Comment:
Hm, this will blockingly wait for the individual entry to be returned. I don't think you want to do this, otherwise every single entry is looked up one by one. It's better to return the StreamIndexChecksum result (like we do for other block reads) and then in the RPC handler for loop over the promises and call `.RetrieveIndexChecksum(..)` one by one (in background will be batched and executed together).